### DISCUSSION TITLE
[Question] Is there a way to prevent multiselect.js from running?

### DISCUSSION THREAD

**Discussion by PhocaCz** [joomla/joomla-cms#37100] [Q&A]
Joomla 4 Administration: When you click on item row, checkbox will be automatically set to checked.

Example: list of articles. When you click on article row, checkbox is selected.

The problem: when the row includes more checkboxes, such checkboxes then are affected too (and it comes to confusing behavior)

media/system/js/multiselect.js

Is there a way to prevent some checkboxes in the row from being affected by multiselect.js (to be ignored by multiselect.js)?

One additional checkbox:
![img](https://i.imgur.com/sSJ9bXJ.gif)

More additional checkboxes:
![img](https://i.imgur.com/9chl434.gif)

Thank you.

---
To avoid unwanted problems, this is a QUESTION, it's not a critique of anyone's work, if you want to answer my question, answer. Thank you.

**Comment by Fedik** [comment]
Remove it from the layout `HTMLHelper::_('behavior.multiselect');`, that should help. 
Currently it does not doing anything useful except checking checboxes "on row click" (bug #25500).

**Comment by brianteeman** [comment]
maybe its possible to restrict the javascript to just a few checkboxes but inthis case i would just remove it completely from the form

**Reply by Fedik** [reply]
That another issue, currently no any restriction. It takes all checkboxes that found in the table:
https://github.com/joomla/joomla-cms/blob/de15a4085b54821abdb9599870fb7a6147e08dbf/build/media_source/system/js/multiselect.es6.js#L17

**Comment by PhocaCz** [comment]
Hi, thank you for your answers.

Yes, removing `HTMLHelper::_('behavior.multiselect');` does the trick. The problem is, then you lose the feature to click on the row and have selected whole row (you need to click directly on the checkbox).

Yes,
```this.boxes = [].slice.call(this.tableEl.querySelectorAll('input[type=checkbox]'));```

I am not Javascript expert but this really looks like it just selects every checkbox it finds. :idea:

**Reply by brianteeman** [reply]
Yes thats exactly what it does.

If it was me I would simply remove the core js from your component and if necessary write something specific for your use.

**Reply by PhocaCz** [reply]
Yes, I think, this will be the best way. Thank you.
